pandas/io/formats/format.py

```diff
@@ -1181,29 +1190,6 @@ def _format_strings(self):
         return fmt_values.tolist()
 
 
-class ExtensionArrayFormatter(GenericArrayFormatter):
-    def _format_strings(self):
-        values = self.values
-        if isinstance(values, (ABCIndexClass, ABCSeries)):
-            values = values._values
-
-        formatter = values._formatter(boxed=True)
```

Comment:
@simonjayhawkins Is this still used in another place? (not very familiar with the formatting code, but wondering where this is called to ensure the underlying can determine the formatting)

Reply:
I don't see it used elsewhere in this PR, or not anymore on master. It might be that we didn't have good tests to cover behaviour where the ExtensionArray deviated from the "normal" behaviour to catch this.

Reply:
> It might be that we didn't have good tests to cover behaviour where the ExtensionArray deviated from the "normal" behaviour to catch this.

i suspect that more tests will need to be added.  particularly with cases where na_rep is passed to to_html etc or 

we do have a specific issue here concerning EAs #25099

categorical, sparse, period etc return None (to defer) or just str (with boxed=True), so the  _formatter was not adding anything.

for datetimelike we have

```
    def _formatter(self, boxed=False):
        # TODO: Remove Datetime & DatetimeTZ formatters.
        return "'{}'".format
```
so will be removed.

for integer array, we have

```
    def _formatter(self, boxed=False):
        def fmt(x):
            if isna(x):
                return 'NaN'
            return str(x)
        return fmt
```

so this is now not being called following the removal of the pre-formatting step. but this likely contributes to the na_rep issues.

IMO the _formatter methods should be used from within the Formatter classes, not passed to them. and cannot assign 'NaN' explcitily.

this should be part a subsequent refactor, see https://github.com/pandas-dev/pandas/pull/26833#discussion_r293530399 and #26837

i should have a follow-on ready shortly continuing the format_array cleanup. (maybe not till next week due to PyLondinium)